bindings/rust/bench/src/harness.rs

```diff
@@ -64,6 +64,9 @@ pub trait TlsBenchHarness: Sized {
 
     /// Get whether or negotiated version is TLS1.3
     fn negotiated_tls13(&self) -> bool;
+
+    /// Transfer given data one-way between connections
+    fn transfer(&mut self, sender: Mode, data: &mut [u8]) -> Result<(), Box<dyn Error>>;
```

Comment:
Why does the data variable need to be mut?

Reply:
Even though everything should return to the same (or very similar at least) state, we still need to mutate the things the `TlsBenchHarness` struct owns in the process, such as the connections and shared buffers.

Reply:
Specifically though, shouldn't the data variable stay un-altered? Like, why would you mutate the data you're sending?

Reply:
Oh sorry, I didn't see that you meant the `data` parameter. It's being mutated because when a connection is receiving things, `data` is given the buffer the connection can write to. With separating out the `send()` and `recv()`, it's also more clear that `send()` doesn't mutate `data` but `recv()` does, and since `transfer()` would have to use `recv()`, it needs `data` as mut.